source/common/router/upstream_request.cc

```diff
@@ -223,6 +224,20 @@ const Network::Connection& UpstreamRequest::connection() const {
   return *parent_.callbacks()->connection();
 }
 
+void UpstreamRequest::copyDynamicMetaDataFromL4Downstream(StreamInfo::StreamInfo& l4_stream_info) {
```

Comment:
Can you explain why this is only implemented for HTTP connection manager and not other network filters? I think it'd make sense for `tcp_proxy` to apply the same logic consistently.

Reply:
I am not sure whether other filters have the similar use case. I think we can add more in future if necessary. In this PR, only focus on the HCM.

Reply:
+1 to at minimum doing this for tcp_proxy

Reply:
I will add.